linux/shim/init.go

```diff
@@ -60,15 +62,27 @@ func newInitProcess(context context.Context, path, namespace string, r *shimapi.
 		}
 		options = *v.(*runcopts.CreateOptions)
 	}
+
+	rootfs := filepath.Join(path, "rootfs")
+	// count the number of successful mounts so we can undo
+	// what was actually done rather than what should have been
+	// done.
+	nrRootMounts := 0
```

Comment:
relatively insignificant, but maybe it's simpler just to do `nrRootMounts = len(r.Rootfs)` here at the top and not worry about counting in the loop?